src/combinatorics/additive/small_roth.lean

```diff
@@ -0,0 +1,137 @@
+/-
+Copyright (c) 2022 Mario Carneiro, Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro, Yaël Dillies, Bhavik Mehta
+-/
+import combinatorics.additive.salem_spencer
+
+/-!
+# Calculation of small Roth numbers
+
+This file implements an algorithm to calculate small Roth numbers. This
```

Comment:
???
Could you sketch the algorithm, and give a link to the main result of the file? (Which would probably deserve a docstring, by the way).